nucleo.cwl

```diff
@@ -0,0 +1,810 @@
+      - id: gatk_collect_alignment_summary_metrics_txt
+    run: uncollapsed_bam_generation/uncollapsed_bam_generation.cwl
+    label: Uncollapsed BAM Generation
+    'sbg:x': 611.546875
+    'sbg:y': 3663.0546875
+  - id: bam_collapsing
+    in:
+      - id: fgbio_group_reads_by_umi_input
+        source: uncollapsed_bam_generation/indel_realignment_bam
+      - id: fgbio_group_reads_by_umi_strategy
+        default: paired
+        source: fgbio_group_reads_by_umi_strategy
+      - id: fgbio_group_reads_by_umi_output_file_name
+        source: fgbio_group_reads_by_umi_output_file_name
+      - id: fgbio_group_reads_by_umi_family_size_histogram
+        source: fgbio_group_reads_by_umi_family_size_histogram
+      - id: fgbio_collect_duplex_seq_metrics_intervals
+        source: fgbio_collect_duplex_seq_metrics_intervals
+      - id: fgbio_collect_duplex_seq_metrics_output_prefix
+        source: fgbio_collect_duplex_seq_metrics_output_prefix
+      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
+        default: true
+        source: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
+      - id: fgbio_call_duplex_consensus_reads_read_group_id
+        source: read-group-id
+      - id: fgbio_call_duplex_consensus_reads_output_file_name
+        source: fgbio_call_duplex_consensus_reads_output_file_name
+      - id: fgbio_call_duplex_consensus_reads_min_reads
+        default:
+          - 1
+          - 1
+          - 0
+        source:
+          - fgbio_call_duplex_consensus_reads_min_reads
+      - id: reference_sequence
+        source: reference_sequence
+      - id: validation_stringency
+        default: LENIENT
+        source: validation_stringency
+      - id: gatk_sam_to_fastq_output_name_R2
+        source: BC_gatk_sam_to_fastq_output_name_R2
+      - id: gatk_sam_to_fastq_output_name_R1
+        source: BC_gatk_sam_to_fastq_output_name_R1
+      - id: bwa_mem_Y
+        default: true
+        source: bwa_mem_Y
+      - id: bwa_mem_T
+        default: 30
+        source: bwa_mem_T
+      - id: sort_order
+        default: coordinate
+      - id: picard_addRG_read_group_sequencing_platform
+        default: ILLUMINA
+        source: platform
+      - id: picard_addRG_read_group_sequencing_center
+        default: MSKCC
+        source: sequencing-center
+      - id: picard_addRG_read_group_run_date
+        source: run-date
+      - id: picard_addRG_read_group_platform_unit
+        source: platform-unit
+      - id: picard_addRG_read_group_library
+        source: library
+      - id: picard_addRG_read_group_identifier
+        source: read-group-id
+      - id: picard_addRG_output_file_name
+        source: BC_picard_addRG_output_file_name
+      - id: bwa_mem_output
+        source: BC_bwa_mem_output
+      - id: create_bam_index
+        default: true
+        source: create_bam_index
+      - id: bwa_mem_K
+        default: 1000000
+      - id: abra2_window_size
+        default: '800,700'
```

Comment:
are there any parameters for `fastp` that need to have defaults here? or maybe we just use the tool defaults

Reply:
@murphycj2  might know if we need to expose anything but I have set a couple of them as defaults

Reply:
@rhshah @ionox0 Yes, based on the default parameters used for trimgalore, I set these parameters for fastp for a comparable trimming result:
- `--length_required 25`
- `--adapter_sequence GATCGGAAGAGC`
- `--adapter_sequence_r2 AGATCGGAAGAGC`
- `--disable_quality_filtering`

Reply:
@murphycj2 thanks I updated the request to have this